extensions/src/main/java/org/pentaho/platform/web/http/filters/ProxyTrustingFilter.java

```diff
@@ -250,16 +285,21 @@ public void doFilter( final ServletRequest request, final ServletResponse respon
 
     if ( ( trustedIpAddrs != null ) && ( request instanceof HttpServletRequest ) ) {
       final HttpServletRequest req = (HttpServletRequest) request;
+      HttpServletResponse res = ( HttpServletResponse ) response;
       String remoteHost = req.getRemoteAddr();
 
       if ( isTrusted( remoteHost ) ) {
         String name = getTrustUser( req );
         if ( !isEmpty( name ) ) {
+          if( csrfValidator != null && !doCsrfValidation( req, res ) ){
```

Comment:
The test for `csrfValidator` being non-null is already done inside `doCsrfValidation`. I think it's best to do it there, not here.